test-utils/src/types.rs

```diff
@@ -114,6 +116,16 @@ impl WebSocketTestServer {
 		Self { local_addr, exit: tx }
 	}
 
+	// Spawns a dummy `JSONRPC v2` WebSocket server that sends out a pre-configured `hardcoded notification` for every connection.
+	pub async fn with_hardcoded_notification(sockaddr: SocketAddr, notification: String) -> Self {
+		let listener = async_std::net::TcpListener::bind(sockaddr).await.unwrap();
```

Comment:
It's `async-std` when I wrote it for a long time ago to avoid using `compat` stuff but we could get rid of that the make the dependency tree a little smaller